packages/graphql/src/classes/utils/asserts-indexes-and-constraints.ts

```diff
@@ -262,55 +262,55 @@ async function getMissingConstraints({
+    const existingConstraints: Map<string, Map<string, string>> = constraintsQueryResult.records
         .map((record) => {
             return record.toObject();
         })
-        .forEach((constraint) => {
-            const label = constraint.labelsOrTypes[0];
-            const property = constraint.properties[0];
-
-            const existingConstraint = existingConstraints[label];
-
-            if (existingConstraint) {
-                existingConstraint.push(property);
-            } else {
-                existingConstraints[label] = [property];
+        .reduce((acc, current) => {
+            const label = current.labelsOrTypes[0];
+            const property = current.properties[0];
+            if (acc.has(label)) {
+                acc.get(label)?.set(property, current.name);
+                return acc;
             }
-        });
+            acc.set(label, new Map([[property, current.name]]));
+            return acc;
+        }, new Map<string, Map<string, string>>());
 
     const missingConstraints: MissingConstraint[] = [];
 
     for (const entity of schemaModel.concreteEntities) {
         const entityAdapter = new ConcreteEntityAdapter(entity);
+        if (!entityAdapter.uniqueFields.length) {
+            continue;
+        }
         for (const uniqueField of entityAdapter.uniqueFields) {
             if (!uniqueField.annotations.unique) {
                 continue;
             }
+            const constraintName = uniqueField.annotations.unique.constraintName;
 
-            let anyLabelHasConstraint = false;
-            for (const label of entity.labels) {
-                // If any of the constraints for the label already exist, skip to the next unique field
-                if (existingConstraints[label]?.includes(uniqueField.databaseName)) {
-                    anyLabelHasConstraint = true;
-                    break;
+            const hasUniqueConstraint = [...entity.labels].some((label) => {
+                const constraintsForLabel = existingConstraints.get(label);
+                if (!constraintsForLabel) {
+                    return false;
                 }
-            }
-
-            if (anyLabelHasConstraint === false) {
-                // TODO: The fallback value of `${entity.name}_${uniqueField.databaseName}` should be changed to use the main label of the entity
-                // But this can only be done once the translation layer has been updated to use the schema model instead of the Node class
-                const constraintName =
-                    uniqueField.annotations.unique.constraintName || `${entity.name}_${uniqueField.databaseName}`;
+                // TODO: remove default value once the constraintName argument is required
+                return (
+                    constraintsForLabel.get(uniqueField.databaseName) ===
+                    (constraintName ?? `${entity.name}_${uniqueField.databaseName}`)
```

Comment:
Could this be a separate variable for better readability?

Reply:
Sure!